kymatio/backend/numpy_backend.py

```diff
@@ -1,24 +1,35 @@
+import numpy
 
 
-class NumpyBaseBackend():
-    def __init__(self, np):
-        self.np = np
+def input_checks(x):
+    if x is None:
+        raise TypeError('The input should be not empty.')
 
-    def input_checks(self, x):
-        if x is None:
-            raise TypeError('The input should be not empty.')
-    
 
-    def complex_check(self, x):
-        if not self._is_complex(x):
-            raise TypeError('The input should be complex.')
-    
+def complex_check(x):
+    if not _is_complex(x):
+        raise TypeError('The input should be complex.')
+
+
+def real_check(x):
+    if not _is_real(x):
+        raise TypeError('The input should be real.')
+
+
+def _is_complex(x):
```

Comment:
Why are these moved outside? Don't they depend on the `np` module we're using (regular NumPy vs. Jax)?

Reply:
We use them in the frontend. Should I just use the back end variable to call the appropriate function in the frontend?

Reply:
I don't understand this answer or its implications.

I guess the question is whether jax can handle numpy dtype objects or not